src/v/raft/compaction_coordinator_rpc_sender.cc

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "raft/compaction_coordinator_rpc_sender.h"
+
+#include "raft/types.h"
+#include "ssx/future-util.h"
+
+#include <type_traits>
+
+namespace raft {
+
+compaction_coordinator_rpc_sender::compaction_coordinator_rpc_sender(
+  ctx_log& logger,
+  vnode self,
+  raft::group_id group,
+  consensus_client_protocol& client_protocol,
+  ss::abort_source& as,
+  ss::gate& bg,
+  compaction_coordinator_rpc_sender::clock_t::duration
+    coordinator_base_interval)
+  : _logger(logger)
+  , _self(self)
+  , _group(group)
+  , _client_protocol(client_protocol)
+  , _raft_as(as)
+  , _raft_bg(bg)
+  , _retry_interval(calculate_retry_interval(coordinator_base_interval)) {}
+
+ss::future<ss::stop_iteration>
+compaction_coordinator_rpc_sender::do_send_local_mcco_to_leader_once(
+  vnode leader_id, model::offset mcco) {
+    if (_raft_as.abort_requested() || _raft_bg.is_closed()) {
+        return ssx::now(ss::stop_iteration::yes);
+    }
+    if (leader_id != _leader_id) {
+        vlog(
+          _logger.trace,
+          "leadership changed from {} to {}, stop sending MCCO={}",
+          leader_id,
+          _leader_id,
+          mcco);
+        // on_leadership_change will re-request once new leader is elected
+        return ssx::now(ss::stop_iteration::yes);
+    }
+    vlog(_logger.trace, "sending MCCO={} to leader={}", mcco, leader_id);
+    return _client_protocol
+      .report_compaction_mcco(
+        leader_id.id(),
+        report_compaction_mcco_request{
+          .node_id = _self,
+          .target_node_id = leader_id,
+          .group = _group,
+          .mcco = mcco},
+        rpc::client_opts{timeout})
+      .then_wrapped([this, mcco, leader_id](
+                      ss::future<result<report_compaction_mcco_reply>>&& fut) {
+          return process_reply(
+            std::move(fut),
+            fmt::format("MCCO={} from leader={}", mcco, leader_id));
+      });
+}
+
+ss::future<ss::stop_iteration>
+compaction_coordinator_rpc_sender::do_send_mtro_to_follower_once(
+  vnode follower_id, model::offset mtro) {
+    if (_raft_as.abort_requested() || _raft_bg.is_closed()) {
+        return ssx::now(ss::stop_iteration::yes);
+    }
+    if (_self != _leader_id) {
+        vlog(
+          _logger.trace,
+          "leader is now {}, stop sending MTRO={} to follower={}",
```

Comment:
nit: in whole `raft` submodule we use `:` instead of `=` to separate variable from its name